src/disco/reducers/installations.js

```diff
@@ -0,0 +1,51 @@
+import {
+  DOWNLOADING,
+  ERROR,
+  INSTALLED,
+  INSTALLING,
+  UNINSTALLED,
+  UNINSTALLING,
+} from 'disco/constants';
+
+export default function installations(state = {}, { type, payload }) {
+  const newState = {...state};
+  if (type === 'INSTALL_STATE') {
```

Comment:
I'm leaving this as `if`/`else if`s since I can omit the `return` in each case which makes things nicer IMO. I added the top-most `if` back in so we aren't copying `state` on every action.